clients/src/main/java/org/apache/kafka/clients/producer/internals/Sender.java

```diff
@@ -338,8 +338,13 @@ private boolean maybeSendTransactionalRequest(long now) {
                 }
 
                 if (targetNode != null) {
-                    if (nextRequestHandler.isRetry())
-                        time.sleep(retryBackoffMs);
+                    if (nextRequestHandler.isRetry()) {
+                        if (0 <= nextRequestHandler.retryBackoffMs())
```

Comment:
Seems like it might be a little simpler to pass the configured retry backoff into `TransactionManager` and use it as the default instead of -1. Then we can remove this check.